### DISCUSSION TITLE
How to screenshot the whole page.

### DISCUSSION THREAD

**Discussion by MahdeenSky** [vgalin/html2image#10] [Q&A]
We have the ability to specify the width and height but what if the website can vary in height everytime we open it, and the same hardcoded size won't work anymore. How do we get around this? To make sure it screenshots the entire site as it will have scroll bars half of the time.

**Comment by vgalin** [comment]
Sadly this is one of the things that cannot be done via Chrome headless in command line (what html2image relies on to take screenshots). I spent some time searching if there were some kind of parameters that allow this but found nothing.
A workaround would be to evaluate the size of the page in some way but I'm not sure if there is a reliable way to do this.

**Reply by bitplane** [reply]
Maybe using a browser extension?

**Comment by jrb-github** [comment]
I am accomplishing this, but the method I'm using has some prerequisites that might not match your situation.  I thought I'd mention it here anyway.

- You can define a fixed width that is reasonable, and a fixed height that is "way too big, but always big enough".
- The background is a flat color.
- Use with the pillow library (pip install pillow) to determine the location of the edge of the real content (find first non-background-color-pixel), provide a border width, and crop.
- You don't care all that much about performance, because my cropping algorithm currently is pretty crude.
